src/sentry/monitors/logic/mark_ok.py

```diff
@@ -33,17 +21,45 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
         "next_checkin": next_checkin,
         "next_checkin_latest": next_checkin_latest,
     }
-    if checkin.status == CheckInStatus.OK:
-        if monitor_env.monitor.status != ObjectStatus.DISABLED:
-            params["status"] = MonitorStatus.OK
-        # in the future this will auto-resolve associated issues
-        if monitor_env.status != MonitorStatus.OK:
-            params["last_state_change"] = ts
-            # resolve any associated incidents
-            incidents = MonitorIncident.objects.filter(
-                monitor_environment=monitor_env, grouphash=monitor_env.incident_grouphash
+
+    # Don't update incidents + certain params if monitor is disabled or already in an OK state
+    if (
+        monitor_env.monitor.status != ObjectStatus.DISABLED
+        and monitor_env.status != MonitorStatus.OK
+    ):
+        params["status"] = MonitorStatus.OK
+        recovery_threshold = monitor_env.monitor.config.get("recovery_threshold", 0)
```

Comment:
Feels like we should just drop this confuging `0` default, it can be 

1. set and is enforced (by the validator) to be above zero
2. set to `None` when the user does not specify the field
3. not set, for old monitors

We should just leave the default blank and let it always be `None`